.github/workflows/CI.yml

```diff
@@ -25,16 +28,52 @@ jobs:
           - ubuntu-latest
         arch:
           - x64
+
     steps:
-      - uses: actions/checkout@v4
-      - uses: julia-actions/setup-julia@v1
+      - uses: actions/checkout@v3
+
+      - name: Set up Julia
+        uses: julia-actions/setup-julia@v1
         with:
           version: ${{ matrix.version }}
           arch: ${{ matrix.arch }}
-      - uses: julia-actions/cache@v1
-      - uses: julia-actions/julia-buildpkg@v1
-      - uses: julia-actions/julia-runtest@v1
-      - uses: julia-actions/julia-processcoverage@v1
-      - uses: codecov/codecov-action@v3
+
+      - name: Cache Julia dependencies
+        uses: julia-actions/cache@v1
+        with:
+          path: ~/.julia
+          key: julia-deps-${{ hashFiles("Project.toml", "Manifest.toml") }}
+
+      - name: Install Julia dependencies
+        run: julia -e 'import Pkg; Pkg.instantiate()'
+
+      - name: Build Julia package
+        uses: julia-actions/julia-buildpkg@v1
+
+      - name: Install PyCall dependencies
+        run: julia -e 'import Pkg; Pkg.add("PyCall"); Pkg.build("PyCall")'
+
+      - name: Run Julia tests
+        uses: julia-actions/julia-runtest@v1
+        with:
+          project: .
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
```

Comment:
do you not need to do this before `PyCall`?

Reply:
I think, PyCall needs Python. Also I'm asking for a specific version - 3.12.